Relationships/RelationshipResolver.cs

```diff
@@ -221,18 +221,36 @@ private void Resolve(CkanModule module, RelationshipResolverOptions options)
                 return;
             }
 
-            foreach (string dep_name in stanza.Select(dep => dep.name))
+            foreach (var descriptor in stanza)
             {
+                string dep_name = descriptor.name;
                 log.DebugFormat("Considering {0}", dep_name);
 
                 // If we already have this dependency covered, skip.
                 // If it's already installed, skip.
-                if (modlist.ContainsKey(dep_name) || registry.IsInstalled(dep_name))
+
+                if (modlist.ContainsKey(dep_name))
                 {
-                    continue;
+                    if (descriptor.version_within_bounds(modlist[dep_name].version))                    
+                        continue;
+                    
+                    //TODO Ideally we could check here if it can be replaced by the version we want. 
+                    throw new InconsistentKraken(string.Format("A certain version of {0} is needed. " +
+                                                               "However a incompatible version is in the resolver", dep_name));
+                    
+                }
+
+                if (registry.IsInstalled(dep_name))
+                {
+                    if(descriptor.version_within_bounds(registry.InstalledVersion(dep_name)))
+                        continue;
+                    //TODO Ideally we could check here if it can be replaced by the version we want. 
+                    throw new InconsistentKraken(string.Format("A certain version of {0} is needed. " +
```

Comment:
For these error messages, is it easy for us to specify what that certain version is? "No" is completely fine, but I know if our users ever see these they'll want to know what version they're supposed to have installed. :)